Framework/API/src/MatrixWorkspace.cpp

```diff
@@ -961,15 +961,15 @@ bool MatrixWorkspace::isHistogramData() const {
  *  @return whether the workspace contains common X bins
  */
 bool MatrixWorkspace::isCommonBins() const {
-  std::lock_guard<std::mutex> lock(m_commonBinsMutex);
-  if (m_isCommonBinsFlag) {
-    return *m_isCommonBinsFlag;
+  const bool isFlagValid{m_isCommonBinsFlagValid.exchange(true)};
```

Comment:
Can you clarify what is happening here? After this line, `m_isCommonBinsFlagValid` should be `true`. Then any other thread will simply return `m_isCommonBinsFlag`, but it may still be in the process of being updated?

Reply:
Yes, or it may be invalidated before the update is finished or another thread enters here. A lock could be added back to this function to prevent at least the scenario you are describing. It is something I have overlooked.